src/Admin/Filter/FilterExecutor.php

```diff
@@ -16,31 +16,39 @@ class FilterExecutor
      */
     public function execute(FilterManager $filterManager, Builder $builder): Builder
     {
-        $filters = $filterManager->getFilters();
         $parameters = $filterManager->getParameters();
 
-        foreach ($filters as $filterItem) {
-            if (! $parameters->has($filterItem->getName())) {
-                continue;
-            }
-
-            // Use user defined executor
-            if ($executor = $filterItem->getExecutor()) {
-                $executor($filterItem, $builder);
+        foreach ($filterManager->getFilters() as $filterItem) {
+            $this->executeForItem($filterItem, $parameters, $builder);
+        }
 
-                continue;
-            }
+        return $builder;
+    }
 
-            $type = $filterItem->getType();
+    /**
+     * @param FilterItem $filterItem
+     * @param FilterParameters $parameters
+     * @param Builder $builder
+     */
+    protected function executeForItem(FilterItem $filterItem, FilterParameters $parameters, Builder $builder): void
```

Comment:
This does not resolve the original comment - this code block still performs 3 different actions